src/index.js

```diff
@@ -147,11 +146,9 @@ export let ContextTree = _.curry(
       }
     }
     let triggerDelayedUpdate = F.debounceAsync(debounce, triggerImmediateUpdate)
-    // Even in immediate update mode, debounce by a very small amount to avoid spamming
-    let triggerSafeImmediateUpdate = F.debounceAsync(10, triggerImmediateUpdate)
```

Comment:
Can we confirm that we really need to roll this back?

Reply:
Yeah it makes the UX a bit worse by adding 10ms delay between opening facet and seeing loading animation, clicking on the search button getting things loading.

Reply:
Can we make it 0 or 1ms then? The debounce would help prevent future bugs where code is calling mutate many times in a row with a self dispatching event

Reply:
Done